File: Roteadores.js
// 1774
const input = require("fs").readFileSync("/dev/stdin", "utf8");
const lines = input.split("\n");

let nextLine = 0;

const [routers, numConnections] = lines[nextLine++].split(" ").map(Number);

let connections = [];
let smallestConnection;

// Similar ao range do python
for (let i in [...Array(numConnections).keys()]) {
  const [n1, n2, l] = lines[nextLine++].split(" ").map(Number);
  const connection = { n1, n2, l };
  if (!smallestConnection || l < smallestConnection.l) {
    smallestConnection = connection;
  }
  connections.push(connection);
}

const nodesConnected = [smallestConnection.n1, smallestConnection.n2];
let totalLength = smallestConnection.l;

while (nodesConnected.length < routers) {
  let closestNode;
  nodesConnected.forEach((node) => {
    const possibleConnections = connections.filter(
      (con) => con.n1 === node || con.n2 === node
    );

    possibleConnections.forEach((possibleConnection) => {
      if (
        !(
          nodesConnected.find(
            (nodeConnected) => nodeConnected === possibleConnection.n1
          ) &&
          nodesConnected.find(
            (nodeConnected) => nodeConnected === possibleConnection.n2
          )
        ) &&
        (!closestNode || possibleConnection.l < closestNode.l)
      ) {
        closestNode = possibleConnection;
      }
    });
  });

  const newNode = nodesConnected.find(
    (connectedNode) => connectedNode === closestNode.n1
  )
    ? closestNode.n2
    : closestNode.n1;
  nodesConnected.push(newNode);
  totalLength += closestNode.l;
}

console.log(totalLength);

File: Roteadores.py
# 1774
routers, num_connections = list(map(int, input().split(" ")))

connections = []

# Vai ser a primeira para começar o sistema
smallest_con = None

for i in range(num_connections):
    n1, n2, l = list(map(int, input().split(" ")))
    con = {'n1': n1, 'n2': n2, 'l': l}
    if not smallest_con or l < smallest_con['l']:
        smallest_con = con
    connections.append(con)

nodes_connected = [smallest_con['n1'], smallest_con['n2']]
total_length = smallest_con['l']

while len(nodes_connected) < routers:
    closest_node = None
    for node in nodes_connected:
        possible_connections = list(filter(
            lambda con: con['n1'] == node 
                     or con['n2'] == node, 
            connections 
        ))

        for p_con in possible_connections:
            # Se ambos os nodos da possível conexão não estão já conctados e sua distância é a menor encontrada até então
            if not (p_con['n1'] in nodes_connected and p_con['n2'] in nodes_connected) and (not closest_node or p_con['l'] < closest_node['l']):
                closest_node = p_con

    new_node = closest_node['n1'] if closest_node['n1'] not in nodes_connected else closest_node['n2']
    nodes_connected.append(new_node)
    total_length += closest_node['l']

print(total_length)